### ISSUE TITLE
Updating 7.5.1 -> 8.0.2 causes `Unable to load class 'org.gradle.api.plugins.MavenPlugin'.` error

### ISSUE THREAD

**Issue by serpro69** [gradle/gradle#24297]
Updated from gradle 7.5.1 to 8.0.2 and can't build the project anymore.

### Expected Behavior

Project should continue building

### Current Behavior

```
Welcome to Gradle 8.0.2!

Here are the highlights of this release:
 - Improvements to the Kotlin DSL
 - Fine-grained parallelism from the first build with configuration cache
 - Configurable Gradle user home cache cleanup

For more details see https://docs.gradle.org/8.0.2/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'kotlin-faker'.
> Failed to notify project evaluation listener.
   > Build completed with 1 failures.
      > org/gradle/api/plugins/MavenPlugin
   > Build completed with 1 failures.
      > org/gradle/api/plugins/MavenPlugin

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.0.2/userguide/command_line_interface.html#sec:command_line_warnings
```

### Context

Worked fined on 7.5.1, stopped working on 8.0.2

### Steps to Reproduce 

- clone https://github.com/serpro69/kotlin-faker
- update gradle to 8.0.2 in https://github.com/serpro69/kotlin-faker/blob/master/gradle/wrapper/gradle-wrapper.properties
- run `./gradlew clean test`

### Your Environment

```
./gradlew --version

------------------------------------------------------------
Gradle 8.0.2
------------------------------------------------------------

Build time:   2023-03-03 16:41:37 UTC
Revision:     7d6581558e226a580d91d399f7dfb9e3095c2b1d

Kotlin:       1.8.10
Groovy:       3.0.13
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          1.8.0_302 (Oracle Corporation 25.302-b08)
OS:           Linux 5.19.0-35-generic amd64
```

**Comment by ov7a** [comment]
Thank you for your interest in Gradle!

Your issue appears to be a question about how to use Gradle. Your question might be answered in Gradle documentation or on our community Slack or forums. Find links to these resources on [help.gradle.org](https://help.gradle.org/).

If you think you have come across a Gradle bug, please file a new issue describing the problem and how to reproduce it.  

If you think Gradle is missing a feature, please file a new issue describing your use case for the feature.

***

Please refer [to the docs](https://docs.gradle.org/current/userguide/upgrading_version_6.html#removal_of_the_legacy_maven_plugin) regarding maven plugin removal in Gradle 7.0. These classes were still available to allow plugins using them to transition. It was completely removed in 8.0+.

**Comment by FyiurAmron** [comment]
tl;dr use

```
plugins {
    id "maven-publish"
}
```

instead of what you had previously, and don't `apply` it afterwards etc., the line above now handles all that.

**Comment by FyiurAmron** [comment]
note that e.g. Spring Boot plugin versions <= 2.4.2 are implicitly trying to use the `maven` plugin anyway, causing Gradle 8.x to break even with `id 'maven-publish'` instead of the `id 'maven'` . Guess this might be the case with some other old/unmaintaned plugins also.

**Comment by Instrumedley** [comment]
@ov7a Terrible answer. The issue is an issue not a question. Faced the same thing, so solution was to go back to previous version

**Comment by ov7a** [comment]
@Instrumedley I'm sorry to hear that, but a major version upgrade [allows](https://semver.org/) to have breaking changes. We [announced this change](https://docs.gradle.org/current/userguide/upgrading_version_6.html#removal_of_the_legacy_maven_plugin) and it is in an upgrade guide.

I'm not sure what answer you expect.

**Comment by serpro69** [comment]
In my case, I haven't been using `maven` plugin either, but some other plugins were.
It has always annoyed me that you can't update gradle because another plugin is using an older version.

**Comment by FyiurAmron** [comment]
@serpro69 I feel you, because I'm in the exact same situation while working with the legacy code in my company, but reasonably speaking, the only thing we can do in such situation is to nag the devs of those plugins to update.

I also agree that the phrasing that "Your issue appears to be a question about how to use Gradle." is, well, rather unfortunate, but I suspect that's it's a canned/autogenerated part of a response. Still, the original link answers the question at least partially: `'maven'` was scheduled to be removed as of 7.x and is no more in 8.x, so any attempt to try to use it, directly or indirectly, will thus fail. Gradle devs can't really help with that, and I think that's quite understandable here also.

**Comment by ov7a** [comment]
@FyiurAmron Thanks for your support!

You're right, it was a canned part and we will improve it to be more suitable for cases similar to that one.

As for the original problem - indeed, its resolution lies on plugin maintainers. We have some integration tests with major plugins to catch potential problems early, but we don't have the capacity to cover all plugins available.
